### ISSUE TITLE
Legend items creeping to the right

### ISSUE THREAD

**Issue by reubengann** [pyqtgraph/pyqtgraph#654]
My code manages the items in the legend manually. For creation
```
vb = pq.ViewBox()
vb.setMaximumWidth(100)
self.ui.theplts.addItem(vb, row = row, col = 1)
legend = pq.LegendItem()
legend.setParentItem(vb)
```
When a new series is plotted, it calls
```legend.addItem(self.pointerToPlot, self.legendtext)```
and when removing 
```legend.removeItem(self.legendtext)```
After creation, I do not interact with the legend in any other way. This works, but if I repeat this process (adding and removing the same item), the legend item moves to the right a couple of pixels each time. If I print legend.layout.geometry(), this effect can be seen
```PyQt5.QtCore.QRectF(0.0, 0.0, 96.0, 120.0)
PyQt5.QtCore.QRectF(0.0, 0.0, 98.0, 120.0)
PyQt5.QtCore.QRectF(0.0, 0.0, 100.0, 120.0)
PyQt5.QtCore.QRectF(0.0, 0.0, 102.0, 120.0)
PyQt5.QtCore.QRectF(0.0, 0.0, 104.0, 120.0)
PyQt5.QtCore.QRectF(0.0, 0.0, 106.0, 120.0)
PyQt5.QtCore.QRectF(0.0, 0.0, 108.0, 120.0)
```

I suspect the issue lies inside the LegendItem.updateSize function, though I do not immediately see a reason for the behavior I observe.

I am using pyqtgraph v 0.10.0 and PyQt 5.6.2 on Windows 7/Anaconda/Python 3.6.4

**Comment by reubengann** [comment]
I have narrowed this down to an increasing sample width. A workaround is

```
legend.addItem(self.pointerToPlot, self.legendtext)
if legend.size is None:
    height = 0
    width = 0
    for sample, label in legend.items:
        height += max(sample.height(), label.height()) + 3
        width = max(width, 11+label.width())
        legend.setGeometry(0,0,width + 25, height)
```

Where I've substituted a constant (11) for the sample.width().

**Comment by reubengann** [comment]
Issue #578 noted this problem.

**Comment by NilsNemitz** [comment]
I think this should have been fixed by #958.
Does anybody still see these issues?

**Comment by j9ac9k** [comment]
I modified the example in `Legend.py` to add/remove/re-add the items, and checked `self.layout.size()` inside the `updateSize` function, and this looks to have been resolved.  I'm going to close out this issue, if it's still problematic, let us know and we'll reopen.  Sorry this has taken us so long to follow up.
